api/invite.go

```diff
@@ -75,6 +80,73 @@ func (a *Api) checkForDuplicateInvite(ctx context.Context, inviteeEmail, invitor
 	return false, nil
 }
 
+//Checks do they have an existing invite or are they already a team member
+//Or are they an existing user and already in the group?
+func (a *Api) checkForDuplicateTeamInvite(ctx context.Context, inviteeEmail, invitorID, teamID, token string, invite models.Type, res http.ResponseWriter) (bool, *shoreline.UserData) {
+
+	//already has invite from this user?
+	invites, _ := a.Store.FindConfirmations(
+		ctx,
+		&models.Confirmation{
+			CreatorId: invitorID,
+			Email:     inviteeEmail,
+			TeamID:    teamID,
+			Type:      invite},
+		[]models.Status{models.StatusPending},
+		[]models.Type{},
+	)
+
+	if len(invites) > 0 {
+		//rule is we cannot send if the invite is not yet expired
+		if !invites[0].IsExpired() {
+			log.Println(statusExistingInviteMessage)
+			log.Println("last invite not yet expired")
+			statusErr := &status.StatusError{Status: status.NewStatus(http.StatusConflict, statusExistingInviteMessage)}
+			a.sendModelAsResWithStatus(res, statusErr, http.StatusConflict)
+			return true, nil
+		}
+	}
+
+	invitedUsr := a.findExistingUser(inviteeEmail, a.sl.TokenProvide())
+	// TODO
+	// call the teams service to check if the user is already a member
```

Comment:
Either create a new method in Crew_client: `GetMember(teamId, userId)` OR reuse the previous call to `GetTeam(teamId)` which should have already returned the list of members

Reply:
I will leverage the GetMember to reduce the number of calls to crew.